django/curator/tag_deduplication.py

```diff
@@ -0,0 +1,264 @@
+import abc
+import logging
+import os
+import re
+import enum
+from typing import List
+
+import dedupe
+from taggit.models import Tag
+
+from curator.models import CanonicalTagMapping, CanonicalTag, TagCluster
+
+
+class AbstractTagDeduper(abc.ABC):
+    TRAINING_FILE = "curator/clustering_training.json"
+    FIELDS = [{"field": "name", "type": "String"}]
+
+    def uncertain_pairs(self):
+        return self.deduper.uncertain_pairs()
+
+    def mark_pairs(self, pairs, is_distinct: bool):
+        labelled_examples = {"match": [], "distinct": []}
+
+        example_key = "distinct" if is_distinct else "match"
+        labelled_examples[example_key] = pairs
+
+        self.deduper.mark_pairs(labelled_examples)
+
+    def prepare_training_data(self):
+        tags = Tag.objects.all().values()
+        data = {row["id"]: row for i, row in enumerate(tags)}
```

Comment:
no need for enumerate here since we're not using the index i, I think `{row["id"]: row for row in Tag.objects.all().values()}` would work just as well?